### ISSUE TITLE
Unable to build library with side-effects modules

### ISSUE THREAD

**Issue by neesonqk** [ng-packagr/ng-packagr#1449]
## Type of Issue
When building library via `ng build my-lib`, all side-effects modules could not be resolved, however, the problem does not exist in development environment.

[*] Bug Report

## Description
Though not recommended practice, but I need some modules to set up some global state that can be used by other modules, those modules are pure JS file, I'm able to run my code in development environment without any problem, however, during building of the library, it complains side-effects modules can not be resolved. 

I have tried that changing `.js` to `.ts` can eliminate this complain, however, my side-effect module is vary large and complicated, which impossible to convert to `.ts`. 

#### How To Reproduce
i.e. within my library, there is a Directive name MyDirective, this directive refers to a side-effect module as follow,

``` ts

import './state.js';

export class MyDirective {

}
```
Above code has no problem with development environment, however, when I run `ng build my-lib`, it starts to complain: "ERROR: Could not resolve './state.js' from dist/my-lib/esm2015/my.directive.js" 

#### Version Information

```
$ node_modules/.bin/ng-packagr --version
ng-packagr: 5.7.1
@angular/*: 8.2.7
typescript: 3.5.3
rxjs: N.A.
node: N.A.
npm/yarn: N.A.
```

**Comment by alan-agius4** [comment]
@neesonqk as you setting the sideEffects field in your package.json?

**Comment by neesonqk** [comment]
@alan-agius4 No, I don't think there is such a field called `sideEffects` in package.json, what I meant is that, import pure JS files to my library, typescript does support this feature, see [this](https://www.typescriptlang.org/docs/handbook/modules.html#import-a-module-for-side-effects-only), and it works well with `ng serve` as well, that is exactly what I need for here, however, it doesn't work with `ng build my-lib`, it complains `ERROR: Could not resolve './...`

**Comment by alan-agius4** [comment]
At the moment this is not possible due to https://github.com/microsoft/TypeScript/issues/7546 which was only addressed in TypeScript 3.7 and is not yet supported by Angular 9.

https://github.com/ng-packagr/ng-packagr/issues/650

**Comment by alan-agius4** [comment]
Let’s keep tracking this here: #650
